src/components/add-wine/useForm.js

```diff
@@ -0,0 +1,61 @@
+import {useState, useEffect, useCallback} from 'react';
+
+function useForm(stateSchema, validationSchema = {}, callback){
+  const [state, setState] = useState(stateSchema);
+  const [disable, setDisable] = useState(true);
+  const [isInvalid, setIsInvalid] = useState(true);
+
+  useEffect(() => {
+    setDisable(true)
+  }, []);
+
+  useEffect(()=>{
+    setDisable(validateState());
+  }, [state, isInvalid]);
+
+  const validateState = useCallback(() => {
+    const hasErrorInState = Object.keys(validationSchema).some(key => {
+      const isInputFieldRequired = validationSchema[key].required;
+      const stateValue = state[key].value;
+      const stateError = state[key].error;
+       return (isInputFieldRequired && !stateValue) || stateError;
+    })
+    return hasErrorInState;
+  }, [state, validationSchema]);
+
+  const handleOnChange = useCallback(event=>{
+    setIsInvalid(true);
+    const name = event.target.name;
+    const value = event.target.value;
+
+    let error = '';
+    if(validationSchema[name].required){
+      if(!value){
+        error = 'This is required field'
+      }
+    }
+
+    if(validationSchema[name].validator !== null && typeof validationSchema[name].validator === 'object'){
```

Comment:
I think that if (validationSchema[name].validator) is sufficient here.

Reply:
Also, can this error-check be extracted into a checkError(schema[key]) function?

Reply:
I have added another condition to check in case there is no validator object for particular field (like wineName)